src/vcpkg/metrics.cpp

```diff
@@ -17,10 +19,92 @@
 #pragma comment(lib, "winhttp")
 #endif
 
+namespace
+{
+    using namespace vcpkg;
+
+    template<typename T>
+    StringLiteral get_metric_name(const T metric, View<MetricEntry<T>> entries)
+    {
+        for (auto&& entry : entries)
```

Comment:
Instead of doing a linear lookup, we should use a unit test to insist that the metrics are exactly ordered by enum value. Then this can become a simple:
```cpp
if (static_cast<size_t>(metric) < entries.size()) {
    return entries[static_cast<size_t>(metric)].name;
} else {
    abort(); // abort() is used because Checks:: will call back into metrics machinery.
}
```

Reply:
done